src/Raven.Server/Documents/Indexes/MapReduce/MapReduceIndexBase.cs

```diff
@@ -165,6 +166,10 @@ protected unsafe int PutMapResults(LazyStringValue lowerId, LazyStringValue sour
 
                             GetResultsStore(reduceKeyHash, indexContext, create: true).Add(mapEntryId, mapResult.Data);
                         }
+
+                        var sourceCollection = CurrentIndexingScope.Current.SourceCollection;
```

Comment:
most cases `TestRun` is null so we don't use `sourceCollection` at all. You can move it directly so:

`AddMapResult(mapResult.Data, AddMapResult(mapResult.Data, sourceCollection);` so it will be called only if needed

Reply:
Changed